packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts

```diff
@@ -240,11 +327,35 @@ export class Safe4337Pack extends RelayKitBasePack<{
    * @return {Promise<SafeOperation>} The Promise object will resolve a SafeOperation.
    */
   async createTransaction({
-    transactions
+    transactions,
+    options = {}
   }: Safe4337CreateTransactionProps): Promise<SafeOperation> {
     const safeAddress = await this.protocolKit.getAddress()
     const nonce = await this.#getAccountNonce(safeAddress)
 
+    if (!this.#paymasterAddress && options?.usePaymaster) {
+      throw new Error('Paymaster address must be initialized')
+    }
+
+    if (options?.erc20TokenAddress) {
+      const { erc20TokenAddress } = options
+
+      const amountToApprove = options.amountToApprove ?? MAX_ERC20_AMOUNT_TO_APPROVE
+
+      const approveToPaymasterTransaction = {
+        to: erc20TokenAddress,
+        data: INTERFACES.encodeFunctionData('approve', [this.#paymasterAddress, amountToApprove]),
+        value: '0',
+        operation: OperationType.Call // Call for approve
+      }
+
+      transactions.push(approveToPaymasterTransaction)
+    }
+
+    const usePaymaster = options?.usePaymaster !== false && !!this.#paymasterAddress
```

Comment:
You already throw an error if `#paymasterAddress` is not defined, why not to just use the option as the flag?

Also you are adding the approve transaction above if they use a `erc20TokenAddress` regardless of the `usePaymaster` option is true or false. 

I think you can improve this code section

Reply:
> You already throw an error if #paymasterAddress is not defined, why not to just use the option as the flag?

The `options?.usePaymaster` flag allows users to choose to use the paymaster for a transaction or not. The check that throws the exception ensures that this flag isn't used without first initializing a `paymasterAddress`.

> Also you are adding the approve transaction above if they use a `erc20TokenAddress` regardless of the usePaymaster option is true or false.

True! I updated the condition to include the `usePaymaster` flag:

```typescript
if (options?.erc20TokenAddress && usePaymaster) {

```